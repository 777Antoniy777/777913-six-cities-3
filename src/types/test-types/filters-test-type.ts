type TestFilter = {
  id: string,
  value: string,
};

type TestFilters = TestFilter[];

export {TestFilter, TestFilters};
