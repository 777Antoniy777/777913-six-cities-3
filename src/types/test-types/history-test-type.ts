type TestRouteHistory = {
  push: () => void;
};

export {TestRouteHistory};
