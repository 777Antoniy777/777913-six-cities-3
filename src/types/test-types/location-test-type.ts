type TestRouteLocation = {
  pathname: string;
};

export {TestRouteLocation};
