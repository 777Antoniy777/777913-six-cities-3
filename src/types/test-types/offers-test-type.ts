type TestCity = {
  id: number;
  name: string;
  coords: number[];
};

type TestHost = {
  avatar: string;
  name: string;
  status: boolean;
};

type TestReview = {
  id: number;
  body: string;
  rating: number;
  name: string;
  date: string;
};

type TestOffer = {
  id: number;
  city: TestCity;
  title: string;
  premium: boolean;
  src: string;
  photos: string[];
  price: number;
  description: string;
  type: string;
  rating: number;
  bedroomAmount: number;
  guestsAmount: number;
  items: string[];
  host: TestHost;
  reviews: TestReview[];
  coord: number[];
};

type TestOffers = TestOffer[];

export {TestOffer};
