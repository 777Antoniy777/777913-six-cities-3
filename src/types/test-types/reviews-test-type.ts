type TestReviewUser = {
  id: number;
  name: string;
  comment: string;
}

type TestReview = {
  id: number;
  user: TestReviewUser;
  status: string;
};

type TestReviews = TestReview[];

export {TestReviews};
