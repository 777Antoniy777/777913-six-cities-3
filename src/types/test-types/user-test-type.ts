type TestUser = {
  id: number;
  name: string;
  avatar: string;
  status: boolean;
};

export {TestUser};
