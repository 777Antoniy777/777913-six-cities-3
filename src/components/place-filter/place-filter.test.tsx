import * as React from "react";
import * as renderer from "react-test-renderer";
import PlaceFilter from "./place-filter";
import {TestFilter, TestFilters} from "../../types/test-types/filters-test-type";

// set mocha data
const isFilterOpened = true;
const currentFilter: TestFilter = {
  id: `1`,
  value: `filter`,
};
const filtersArr: TestFilters = [
  {
    id: `1`,
    value: `filter`,
  },
];

const setFilterStatus = () => ({});
const getCurrentFilter = () => ({});

it(`render PlaceFilter`, () => {
  const tree = renderer.create(
      <PlaceFilter
        isFilterOpened={isFilterOpened}
        currentFilter={currentFilter}
        filtersArr={filtersArr}
        setFilterStatus={setFilterStatus}
        getCurrentFilter={getCurrentFilter}
      />)
      .toJSON();

  expect(tree).toMatchSnapshot();
});
