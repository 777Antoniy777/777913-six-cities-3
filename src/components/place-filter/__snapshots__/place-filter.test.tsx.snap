// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PlaceFilter 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
    onClick={[Function]}
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex={0}
  >
    filter
    <svg
      className="places__sorting-arrow"
      height={4}
      width={7}
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex={0}
    >
      filter
    </li>
  </ul>
</form>
`;
